telemetry/istio/src/main/scala/io/buoyant/telemetry/istio/MixerClient.scala

```diff
@@ -0,0 +1,114 @@
+package io.buoyant.telemetry.istio
+
+import com.google.local.DurationProto.Duration
+import com.twitter.finagle.buoyant.h2
+import com.twitter.finagle.Service
+import com.twitter.util.Future
+import istio.mixer.v1.{Attributes, Mixer, ReportRequest, ReportResponse, StringMap}
+
+private[telemetry] object MixerClient {
+
+  // TODO: build up this dictionary as requests are sent out,
+  // also consider using defaultWords, once per ReportRequest, rather than per Attributes
+  private[telemetry] val words = Seq[String](
```

Comment:
There's no need to repeat the well-known attribute names in your `words` list: they already exist in the global list, so you can index into that list for their names. Then, you only need to include your own custom words in your per-request list. That should cut down on a lot of data over the wire.

The global list is defined here: https://github.com/istio/api/blob/master/mixer/v1/global_dictionary.yaml

You'll want to include the [global word count](https://github.com/istio/api/blob/master/mixer/v1/report.proto#L56) with each of your requests anyway (IIRC Mixer will reject the call if it can't agree on a dictionary), so you need either a copy of the global list or to hard-code the number of entries anyway. We do not anticipate changing the global list frequently, and any change made to it will be a breaking API change, so there will be plenty of warning. [We embed it directly into the Mixer.](https://github.com/istio/mixer/blob/master/pkg/api/globalWordList.go)